src/test/java/com/commercetools/sync/inventories/InventorySyncTest.java

```diff
@@ -143,20 +150,73 @@ public void sync_WithEnsuredChannels_ShouldCreateEntriesWithUnknownChannels() {
         assertThat(stats.getCreated()).isEqualTo(1);
         assertThat(stats.getFailed()).isEqualTo(0);
         assertThat(stats.getUpdated()).isEqualTo(0);
+        assertThat(errorCallBackMessages).hasSize(0);
+        assertThat(errorCallBackExceptions).hasSize(0);
     }
 
     @Test
     public void sync_WithNotEnsuredChannels_ShouldNotSyncEntriesWithUnknownChannels() {
         final InventoryEntryDraft draftWithNewChannel = InventoryEntryDraft.of(SKU_3, QUANTITY_1, DATE_1, RESTOCKABLE_1,
                 Channel.referenceOfId(KEY_3));
-        final InventorySync inventorySync = getInventorySync(30, false);
+
+        final InventorySyncOptions options = getInventorySyncOptions(30, false, true);
+        final InventoryService inventoryService = getMockInventoryService(existingInventories,
+            mock(InventoryEntry.class), mock(InventoryEntry.class));
+        final ChannelService channelService = mock(ChannelService.class);
+        when(channelService.fetchCachedChannelIdByKeyAndRoles(anyString(), any()))
+            .thenReturn(completedFuture(Optional.empty()));
+
+        final InventorySync inventorySync = new InventorySync(options, inventoryService, channelService,
+            mock(TypeService.class));
+
         final InventorySyncStatistics stats = inventorySync.sync(singletonList(draftWithNewChannel))
                 .toCompletableFuture()
                 .join();
         assertThat(stats.getProcessed()).isEqualTo(1);
         assertThat(stats.getFailed()).isEqualTo(1);
         assertThat(stats.getCreated()).isEqualTo(0);
         assertThat(stats.getUpdated()).isEqualTo(0);
+        assertThat(errorCallBackMessages).hasSize(1);
+        assertThat(errorCallBackMessages.get(0)).isEqualTo(format("Failed to resolve supply channel reference on"
+            + " InventoryEntryDraft with sku:'%s'. Reason: "
+            + "com.commercetools.sync.commons.exceptions.ReferenceResolutionException: "
+            + "Channel with key '%s' does not exist.", SKU_3, KEY_3));
```

Comment:
Shouldn't be constants resued somehow in such comparisions? Let's imagine someone change `ReferenceResolutionException` name or move it to another package, how many tests would this affect?